test/FastTests/Client/QueriesWithCustomFunctions.cs

```diff
@@ -1577,6 +1577,73 @@ select new
         }
 
         [Fact]
+        public void Custom_Functions_ValueTypeParse_Support()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User { Name = "Jerry", LastName = "Garcia" }, "users/1");
+                    session.Store(new User { Name = "Phil", LastName = "" }, "users/2");
+                    session.Store(new User { Name = "Pigpen" }, "users/3");
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = from u in session.Query<User>()
+                                select new
+                                {
+                                    IntParse = int.Parse("1234"),
+                                    DoubleParse = double.Parse("1234"),
+                                    DecimalParse = decimal.Parse("1234"),
+                                    BoolParse = bool.Parse("true"),
+                                    CharParse = char.Parse("s"),
+                                    ByteParse = byte.Parse("127"),
+                                    LongParse = long.Parse("1234"),
+                                    SByteParse = sbyte.Parse("127"),
+                                    ShortParse = short.Parse("1234"),
+                                    UintParse = uint.Parse("1234"),
+                                    UlongParse = ulong.Parse("1234"),
+                                    UshortParse = ushort.Parse("1234")
+                                };
+
+                    Assert.Equal("FROM Users as u SELECT { " +
+                        "IntParse : \"1234\", " +
```

Comment:
But this outputs a _string_, not a value. 
That is probably not a good idea.
We want to send the raw value, not as a string.